src/module.cpp

```diff
@@ -2805,8 +2850,11 @@ int Module::CompileAndOutput(const char *srcFile, Arch arch, const char *cpu, st
             return 1;
 
         m = new Module(srcFile);
-        if (m->CompileFile() == 0) {
-            llvm::TimeTraceScope TimeScope("Backend");
+        const int compileResult = m->CompileFile();
+
+        llvm::TimeTraceScope TimeScope("Backend");
+
+        if (compileResult == 0) {
```

Comment:
> why did you do this change?
> 
> const int compileResult = m->CompileFile(); llvm::TimeTraceScope TimeScope("Backend"); if (compileResult == 0) {
> 
> the difference that I see is that only the scope that TimeScope("Backend"); covers
> 
> and it now covers a couple delete operators. I'm not sure if it's intended
> 
> I.e. in covers the timing between its creation till destruction. By pulling it to another scope you extended its live time. So the question if this is intended or not
@dbabokin 

In the previous version of `CompileAndOutput`, when `! (targets.size() == 0 || targets.size() == 1)`, on every path processing `m`, `Timescope` begins after the call to `CompileFile` and captures the timing for the check of the result of that call.

The change for the `(targets.size() == 0 || targets.size() == 1)` path done here brings the measurement in-line with the other path. This was the easiest way to make the measurement match on both paths, but another structure could also work.

Reply:
That's a good catch, we need to unify it for single-target and multi-target compilations. I think the changes you did make sense. Thanks for explainaing.